### PR TITLE
feat: guardian module refactoring

### PR DESCRIPTION
**Pull request by Eddort**

### CODE REVIEW

**Review by avsetsin — APPROVED** [review]
lgtm

**Review by infloop — APPROVED** [review]

File: src/guardian/block-guard/block-guard.service.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      const guardianAddress = this.securityService.getGuardianAddress();
+
+      const [depositRoot, depositedEvents, guardianIndex] = await Promise.all([
+        this.depositService.getDepositRoot({ blockHash }),
+        this.depositService.getAllDepositedEvents(blockNumber, blockHash),
+        this.securityService.getGuardianIndex({ blockHash }),
+        this.depositService.handleNewBlock(blockNumber),
+      ]);
+
+      endTimer();
```
Comment (avsetsin) [inline src/guardian/block-guard/block-guard.service.ts:78]: should be called in `finally` section

File: src/guardian/guardian-metrics/guardian-metrics.service.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  /**
+   * Collects metrics about operators keys
+   * @param blockData - collected data from the current block
+   */
+  public collectOperatorMetrics(stakingModuleData: StakingModuleData): void {
+    const { unusedKeys, stakingModuleId } = stakingModuleData;
+
+    const operatorsKeysTotal = unusedKeys.length;
+    this.operatorsKeysCounter.set(
+      { type: 'total', stakingModuleId },
```
Comment (avsetsin) [inline src/guardian/guardian-metrics/guardian-metrics.service.ts:104]: not total, but unused

File: src/guardian/guardian-metrics/guardian-metrics.service.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      depositedKeys.length,
+    );
+    this.depositedKeysCounter.set(
+      { type: 'unique', stakingModuleId },
+      depositedKeysSet.size,
+    );
+    this.depositedKeysCounter.set(
+      { type: 'duplicates', stakingModuleId },
+      depositedDubsTotal,
```
Comment (avsetsin) [inline src/guardian/guardian-metrics/guardian-metrics.service.ts:91]: I guess we can get rid of the `duplicates`. It was not useful.

File: src/guardian/staking-module-guard/staking-module-guard.spec.ts [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+      nextSigningKeys: [] as string[],
+      nodeOperatorsCache,
+      depositedEvents,
+      guardianAddress: '0x3456',
+      guardianIndex: 1,
+      isDepositsPaused: false,
+      srModuleId: 1,
+    };
+
+    it('should call handleKeysIntersections if next keys are found in the deposit contract', async () => {
```
Comment (avsetsin) [inline src/guardian/staking-module-guard/staking-module-guard.spec.ts:158]: next keys -> unused keys

File: src/guardian/staking-module-guard/staking-module-guard.spec.ts [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+      expect(mockHandleCorrectKeys).not.toBeCalled();
+      expect(mockHandleKeysIntersections).toBeCalledTimes(1);
+      expect(mockHandleKeysIntersections).toBeCalledWith(
+        { ...stakingModuleData, unusedKeys },
+        blockData,
+      );
+      expect(mockGetWithdrawalCredentials).toBeCalledTimes(1);
+    });
+
+    it('should call handleCorrectKeys if Lido next keys are not found in the deposit contract', async () => {
```
Comment (avsetsin) [inline src/guardian/staking-module-guard/staking-module-guard.spec.ts:197]: next keys -> unused keys

**Pull request merged**